File: backend/Models/ZohoProject.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("ZohoProjects")]
    public class ZohoProject
    {
        [Key]
        public string? ProjectId { get; set; }
        public string? ProjectName { get; set; }
        public string? ClientId { get; set; }
        public string? ClientName { get; set; }
        public string? ProjectStatus { get; set; }
        public string? OwnerId { get; set; }
        public string? OwnerName { get; set; }
        public bool? IsDeleteAllowed { get; set; }

        public string? ProjectManagers { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/app/master/models/ZohoProject.ts
export interface ProjectManager {
  name: string;
  rate: number;
  empId: string;
  erecno: string;
}

export interface ZohoProject {
  // WSR-Specific Fields — update these from `never[]` to their actual structure
  plannedActivities?: { task: string; status: string; remarks: string; planned: boolean }[];
  progressData?: { task: string; status: string; remarks: string; planned: boolean }[];
  keyIssues?: {
    type: string;
    functionalArea: string;
    description: string;
    actions: string;
    dateRaised: string;
    resolveBy: string;
    owner: string;
  }[];
  keyRisks?: {
    description: string;
    mitigation: string;
    likelihood: string;
    owner: string;
    dateRaised: string;
    resolveBy: string;
  }[];

  // Existing Fields
  currencyType: string;
  customerId: string;
  projectId: string;
  projectName: string;
  startDate: string;
  endDate: string;
  clientId: string;
  clientName: string;
  projectStatus: string;
  ownerId: string;
  ownerName: string;
  isDeleteAllowed: boolean;
  projectManagers: string;
  createdOn: string;

  // Optional: For form submission enhancements
  projectType?: string;
  projectDescription?: string;
}
